source/common/http/status.h

```diff
@@ -109,5 +109,36 @@ ABSL_MUST_USE_RESULT bool isCodecClientError(const Status& status);
  */
 Http::Code getPrematureResponseHttpCode(const Status& status);
 
+/**
+ * Macro that checks return value of expression that results in Status and returns from
+ * the current function is status is not OK.
+ *
+ * Example usage:
+ *   Status foo() {
+ *     RETURN_IF_ERROR(bar());
+ *     return okStatus();
+ *   }
+ */
+
+#define RETURN_IF_ERROR(expr)                                                                      \
+  if (::Envoy::Http::Details::StatusAdapter adapter{(expr)}) {                                     \
+  } else                                                                                           \
+    return std::move(adapter.status_)
```

Comment:
nit: full braces around the else statement please.

Reply:
But then you will not be able to put `;` at the end of RETURN_IF_ERROR without clang-tidy complaining.

Reply:
What if you wrap the entire thing in a brace scope? Does that work?

Reply:
Hmm, the clang-tidy error is rather weird. I'm unsure what to make of it. Here is the latest change.

```
error: invalid argument '-std=c++14' not allowed with 'C' [clang-diagnostic-error]
```

Reply:
This happens when the clang-tidy bazel rules are messed up. @lizan will know how to fix.